### PR TITLE
Fix OCR KTP Camera and Photo Compression

### PR DESCRIPTION
**Pull request by gregoriusjimmy**
## Type of changes

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

- Cam compressed photo after taking a pic
- Cam aspect ratio on desktop  is not same as on mobile devices
- KTP OCR on mobile can not be processed because of Cam width & height less than 400 x 300
## What is the new behavior?

- Fix width and height Cam using mediaQuery hooks
- Photo quality set to 1 instead of default 0.96
- Disable pixel smoothing
- Cam aspect ratio on mobile now is 1:1 and 4:3 for desktop
## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

image
image
image

### CODE REVIEW

File: frontend/app/components/modules/Cam/Cam.tsx [hunk 1]
```diff
@@ -10,26 +10,32 @@ import {
...
 import { Button } from '../../elements/Button/Button'
 import { Color } from '../../../types/elements'
+import { useMediaQuery } from 'app/hooks/useMediaQuery'
 
 type Props = {
   localkey: string
   nextStep: MouseEventHandler<HTMLButtonElement>
-  videoConstraints?: Object
   overlayShape: 'rect' | 'circle'
+  videoConstraints?: Object
```
Comment (krismp) [inline frontend/app/components/modules/Cam/Cam.tsx:19]: Object nya attribute nya fixed gak @gregoriusjimmy klo fixed define type aja

File: frontend/app/components/modules/Cam/Cam.tsx [hunk 2]
```diff
@@ -67,7 +73,22 @@ export const Cam = ({
               audio={false}
               ref={webcamRef}
               screenshotFormat="image/jpeg"
-              videoConstraints={videoConstraints}
+              screenshotQuality={1}
+              videoConstraints={
+                videoConstraints
+                  ? videoConstraints
+                  : {
+                      width: { min: isMobile ? 420 : 450 },
```
Comment (krismp) [inline frontend/app/components/modules/Cam/Cam.tsx:81]: objectnya cuma `min` aja? gak langsung aja `width: n`

Comment (gregoriusjimmy) [reply frontend/app/components/modules/Cam/Cam.tsx:81]: biar responsive ngikutin width container stylenya kalau dimobile. udah di cek gapake `min` kameranya gamau kebuka pas di mobile

File: frontend/app/components/modules/Cam/Cam.tsx (hunk 1 above)
Comment (gregoriusjimmy) [reply frontend/app/components/modules/Cam/Cam.tsx]: done @krismp

Comment (krismp) [reply frontend/app/components/modules/Cam/Cam.tsx]: Type nya gak ke commit jim?

Comment (gregoriusjimmy) [reply frontend/app/components/modules/Cam/Cam.tsx]: type global itu ka https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints, udah ada di PR eslintnya fu juga

Comment (krismp) [reply frontend/app/components/modules/Cam/Cam.tsx]: TIL baru tw

**Review by krismp — COMMENTED** [review]
lgtm

**Review by krismp — APPROVED** [review]

**Review by williammfu — APPROVED** [review]
🤩

**Pull request merged**